### PR TITLE
bpo-39793: use the same domain on make_msgid tests

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39793

### CODE REVIEW

File: Lib/test/test_email/test_email.py [hunk 1]
```diff
@@ -3342,9 +3342,10 @@ def test_make_msgid_idstring(self):
             '.test-idstring@testdomain-string>')
 
     def test_make_msgid_default_domain(self):
```
Comment (akuchling) [inline Lib/test/test_email/test_email.py:3344]: Unfortunately this change removes the whole point of the test, which is to check what happens when the domain parameter *isn't* being provided.  I think instead the test should just mock socket.getfqdn(), something like this (untested by me):
```
import mock   # at top of module

   with mock.patch('socket.getfqdn') as mock_getfqdn:
       mock_getfqdn.return_value = domain = 'pythontest.example.com'
       self.assertTrue( ... ) 
```

But I do think this is worth fixing.

**Review by akuchling — CHANGES_REQUESTED** [review]

File: Lib/test/test_email/test_email.py (hunk 1 above)
Comment (isidentical) [reply Lib/test/test_email/test_email.py:3344]: Sorry for my misunderstanding over tests concept. Will apply your suggestion, thanks.

**Comment by isidentical** [comment]
I have made the requested changes; please review again

File: Lib/test/test_email/test_email.py (hunk 1 above)
Comment (akuchling) [reply Lib/test/test_email/test_email.py:3344]: The new version looks good to me; thanks!

**Review by akuchling — APPROVED** [review]

**Comment by akuchling** [comment]
@python/email-team -- do you want to veto this PR?  Otherwise I'll go ahead and merge it.

**Comment by isidentical** [comment]
@akuchling no objection so far, 12 days.

**Pull request merged**